spec/requests/api_requests_spec.rb

```diff
@@ -0,0 +1,27 @@
+require 'rails_helper'
+describe "get index request " do 
+  before(:example) do |example|
+    @user = create(:user)
+    @apikey = Apikey.create(user: @user, access_key: SecureRandom.hex)
+  end
+  it "creates a todo" do 
+    post '/api/v1/todos/?todo[description]="hello world"', nil, {'HTTP_AUTHORIZATION' => "Token token=#{@apikey.access_key}"}
+    expect(response.status).to eq(201)
+    expect(@user.todos.count).to eq(1)
+  end
+  it "gets a list of todo items" do
+    todo = create(:todo, description: "helloworld", user: @user, id: 1)
+    expect(@user.todos.count).to eq(1)
+    get '/api/v1/todos', nil, {'HTTP_AUTHORIZATION' => "Token token=#{@apikey.access_key}"}
+    expect(response.status).to eq(200) 
+    expect(response.body).to eq(@user.todos.to_json)
+    json_response = response.body
+  end
+  it "deletes a todo" do
+    todo = create(:todo, description: "helloworld", user: @user, id: 1)
```

Comment:
Avoid hardcoding ids like this when creating objects in tests. Create the object and then call `.id` on it if you need it later in the test.

Line 21 would become:
```ruby
  todo = create(:todo, description: "helloworld", user: @user)
```

and line 23 would use string interpolation in the URL path:
```ruby
delete "/api/v1/todos/#{todo.id}", nil, {'HTTP_AUTHORIZATION' => "Token token=#{@apikey.access_key}"}
```